### ISSUE TITLE
Quantization Aware Training On Eager Mode

### ISSUE THREAD

**Issue by vibhatha** [tensorflow/tensorflow#31034]
I am developing my model in the eager mode by using Keras models. 
I read a couple of examples and guidelines in quantization aware training. 
Most of them are not using gradient tape but uses the inbuilt training function
to train and it looks straight forward. But let's say, someone, is calculating the gradient and have written the training model in a custom way. 
How to use quantization aware training in such a case?
I am asking this question because the function call takes input from a created tf.Graph() instance. If someone has tf.GradientTape() instance, how to deal with this?

I am new to Tensorflow concepts, maybe I am not following it correctly.  It will be great if someone can explain to me how to use quantization aware training in general.

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
